### PR TITLE
Render as events

### PR DESCRIPTION
**Pull request by vincelugli**
Add structure to support #3. Wraps on resize by events instead of voice.

- Add offset (number of characters from previous event to put content) to voices in events. Offset is 1 + difference between start of event and voice index. Split content sets offset to 0 for no space.
- Rename `EventList` to `Line` (let me know if `EventList` is more clear, just felt like `event` was showing up all over the place).
  - Verse -> Phrase -> Line -> Event -> Voice
- Move `VoiceColors` to new file.
- Move adding diagram to chord div to new file.
- Create new renderer to handle events.
- Update default renderer to event based one.
- Emphasize chords that are not in the library.

## Resize and wrap
render_events

## Emphasize chords missing from library
image

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ function MarkdownMusic(md, musicOpts) {
     highlight: function(str, lang) {
       if (md.highlightRegistry.hasOwnProperty(lang)) {
         try {
-          return md.highlightRegistry[lang](str, md.musicOpts);
+          return `<pre style="display: none;"></pre>${md.highlightRegistry[lang](str, md.musicOpts)}`;
```
Comment (shadanan) [inline index.js:14]: Add comment for why we do this stupidness.

File: parsers/events.js [hunk 2]
```diff
@@ -62,11 +64,12 @@ class EventList {
    *
    * @param {Map<String, Object[]>} phrase The phrase to pick events from.
    * @param {Object} longestEvent The longest (by content length) and first voice event from phrase.
+   * @param {number} startIndex The index for the start of this event.
    * @return {String[]} voiceAdded Voice names added to event list.
-   * @return {Object[]} eventList List of the first of each voice from phrase that fit within bounds of longest event.
+   * @return {Object[]} line List of the first of each voice from phrase that fit within bounds of longest event.
```
Comment (shadanan) [inline parsers/events.js:69]: I used to call this a segment. It's essentially an indivisible (doesn't internally wrap) thing, like:

```
Am
Brown
```

Is that what you mean by this?

File: parsers/events.spec.js [hunk 3]
```diff
@@ -173,33 +173,33 @@ describe('Event', () => {
...
+          { index: 0, offset: 1, voice: 'l1', content: 'longest' },
         ],
         [
-          { index: 8, voice: 'l1', content: 'is' },
+          { index: 8, offset: 1, voice: 'l1', content: 'is' },
         ],
         [
-          { index: 11, voice: 'c1', content: new Chord('C') },
-          { index: 11, voice: 'l1', content: 'supercali' }
+          { index: 11, offset: 1, offset: 1, voice: 'c1', content: new Chord('C') },
```
Comment (shadanan) [inline parsers/events.spec.js]: duplicate?

**Review by shadanan — APPROVED** [review]
LGTM!

File: parsers/events.js (hunk 2 above)
Comment (vincelugli) [reply parsers/events.js:69]: I meant more of a line of a phrase, or a list of segments. For example,

```
c1:                    Am     G  F          G      Esus4  E
v1: All the leaves are brown        and the sky is gray

c1: F               C     E  Am       F        Esus4  E
v1: I've been for a walk         on a winter's day

c1:                 Am    G  F       G      Esus4  E
v1: I'd be safe and warm        if I was in L.A.

c1:            Am        G  F     G               Esus4  E
v1: California dreamin'        on such a winter's day
```

a line would be 

```

                   Am     G  F          G      Esus4  E
All the leaves are brown        and the sky is gray
```

**Pull request merged**